packages/firestore/test/unit/specs/existence_filter_spec.test.ts

```diff
@@ -351,25 +510,56 @@ describeSpec('Existence Filters:', [], () => {
   specTest('Bloom filter limbo resolution is denied', [], () => {
     const query1 = query('collection');
     const docA = doc('collection/a', 1000, { v: 1 });
-    const docB = doc('collection/b', 1000, { v: 2 });
+    const docB = doc('collection/b', 1000, { v: 1 });
+    const bloomFilterProto = generateBloomFilterProto({
+      contains: ['collection/a'],
+      notContains: ['collection/b'],
+      numOfBits: 2,
+      hashCount: 1
+    });
     return spec()
       .userListens(query1)
       .watchAcksFull(query1, 1000, docA, docB)
       .expectEvents(query1, { added: [docA, docB] })
-      .watchFilters([query1], [docA.key], {
-        bits: { bitmap: 'AQ==', padding: 6 },
-        hashCount: 1
-      })
+      .watchFilters([query1], [docA.key], bloomFilterProto)
       .watchSnapshots(2000)
       .expectEvents(query1, { fromCache: true })
-      .expectLimboDocs(docB.key) // docB is now in limbo
+      .expectLimboDocs(docB.key) // docB is now in limbo.
       .watchRemoves(
         newQueryForPath(docB.key.path),
         new RpcError(Code.PERMISSION_DENIED, 'no')
       )
-      .expectLimboDocs() // docB is no longer in limbo
+      .expectLimboDocs() // docB is no longer in limbo.
       .expectEvents(query1, {
         removed: [docB]
       });
   });
+
+  specTest('Bloom filter with large size works as expected', [], () => {
+    const query1 = query('collection');
+    const docs = [];
+    for (let i = 0; i < 100; i++) {
+      docs.push(doc(`collection/doc${i}`, 1000, { v: 1 }));
+    }
+    const docKeys = docs.map(item => item.key);
+
+    const bloomFilterProto = generateBloomFilterProto({
+      contains: docKeys.slice(0, 50).map(item => item.toString()),
+      notContains: docKeys.slice(50).map(item => item.toString()),
+      numOfBits: 1000,
+      hashCount: 16
+    });
+    return (
+      spec()
+        .userListens(query1)
+        .watchAcksFull(query1, 1000, ...docs)
+        .expectEvents(query1, { added: docs })
+        // Doc0 to doc50 are deleted in the next sync.
```

Comment:
This comment is incorrect: "Doc0 to doc50 are deleted" because "doc50" is _not_ deleted. Maybe instead use a comment like this: "Documents 0 to 49, inclusive, are deleted".